tests/middlewareTypes.test.tsx

```diff
@@ -19,6 +19,27 @@ type CounterState = {
   inc: () => void
 }
 
+type ExampleStateCreator<T, A> = <
+  Mps extends [StoreMutatorIdentifier, unknown][] = [],
+  Mcs extends [StoreMutatorIdentifier, unknown][] = [],
+  U = T,
+>(
+  f: StateCreator<T, [...Mps, ['org/example', A]], Mcs>,
+) => StateCreator<T, Mps, [['org/example', A], ...Mcs], U & A>
+
+type Write<T, U> = Omit<T, keyof U> & U
+type StoreModifyAllButSetState<S, A> = S extends {
+  getState: () => infer T
+}
+  ? Omit<StoreApi<T & A>, 'setState'>
```

Comment:
I'm not sure if I understand this fully, but omitting setState may prevent inferring type from setState, no?
The reason the immer impl works is it's using `set` instead of `setState`, but we don't get types from `set`.
I'm a bit confused what's going on.

Reply:
Sorry, just updated that. The goal is to use an example middleware that modifies the store state to add some `A` type to all *but* the `setState`, and then assert that it affected the types appropriately on both the inside of the Immer middleware and the outside.

Reply:
yeah, but still it doesn't feel like a good example, because the implementation of the example middleware should include `delete api.setState`, but `immer` middleware infers types from `setState`.

Reply:
Why would the implementation need to include `delete api.setState`? The example rewrites `setState`, it doesn't delete it. 

Do you agree that the type assertions are sufficient? Since this change is strictly on a type level, I didn't think implementation was necessary.

Reply:
> it doesn't delete it.

Okay, I think I misread something then.

Reply:
yeah, I think I forgot that `Write` is an intersection.
Never mind.